jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/LoggingMXBeanImpl.java

```diff
@@ -151,20 +206,61 @@ public String getParentLoggerName(String loggerName) {
 	 */
 	@Override
 	public void setLoggerLevel(String loggerName, String levelName) {
-		final Logger logger = LogManager.getLogManager().getLogger(loggerName);
-
+		final Object logger;
+		try {
+			logger = getLoggerFromName(loggerName);
+		} catch (Exception e) {
+			throw newInternalInvokeError(e);
+		}
+		
 		if (logger != null) {
 			// The named Logger exists. Now attempt to set its level. The
 			// below attempt to parse a Level from the supplied levelName
 			// will throw an IllegalArgumentException if levelName is not
 			// a valid level name.
-			Level newLevel = Level.parse(levelName);
-			logger.setLevel(newLevel);
+			try {
+				Object newLevel = le_parse.invoke(null, levelName);
+				lo_setLevel.invoke(logger, newLevel);
+			} catch (Exception e) {
+				throw newInternalInvokeError(e);
+			}
+			
 		} else {
 			// Named Logger does not exist.
 			/*[MSG "K05E7", "Unable to find Logger with name {0}."]*/
 			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05E7", loggerName)); //$NON-NLS-1$
 		}
 	}
-
+	
+	private static Object getLoggerFromName(String loggerName) throws Exception {
+		Object logManagerInstance = lm_getLogManager.invoke(null);
+		return lm_getLogger.invoke(logManagerInstance, loggerName);
+	}
+	
+	/* Handle error thrown by method invoke as internal error */
+	private static InternalError newInternalInvokeError(Exception cause){
+		handleInvocationTargetException(cause);
+		
+		InternalError err = new InternalError(cause.toString());
+		err.initCause(cause);
+		return err;
```

Comment:
this should `throw err`, not return to match the behaviour of `handleInvocationTargetException()` and to ensure the caller can't forgot to throw the Exception